AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/DataStoreConfiguration.swift

```diff
@@ -0,0 +1,103 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+
+/// Error Handler function typealias
+public typealias DataStoreErrorHandler = (AmplifyError) -> Void
+
+/// Holds a reference to both the local `Model` and the remote one during a conflict
+/// resolution. Implementations of the `DataStoreConflictHandler` use this to decide
+/// what the outcome of a conflict should be.
+public struct DataStoreConclictData {
+    public let local: Model
+    public let remote: Model
+}
+
+/// Conflict Handler function typealias. The function is used during a conflict that
+/// could not be resolved and requires a decision from the consumer.
+public typealias DataStoreConflictHandler = (DataStoreConclictData, DataStoreConflictHandlerResolver) -> Void
+
+/// Callback for the `DataStoreConflictHandler`.
+public typealias DataStoreConflictHandlerResolver = (DataStoreConflictHandlerResult) -> Void
+
+/// The conflict resolution result enum.
+public enum DataStoreConflictHandlerResult {
+    
+    /// Discard the local changes in favor of the remote ones.
+    case discard
+    
+    /// Keep the local changes.
+    case keep
+    
+    /// Return a new `Model` instance that should used instead of the local and remote changes.
+    case new(Model)
+}
+
+
+/// The `DataStore` plugin configuration object.
+public struct DataStoreConfiguration {
+
+    public let errorHandler: DataStoreErrorHandler
+    public let conflictHandler: DataStoreConflictHandler
+    public let syncInterval: UInt
+    public let syncMaxRecords: UInt
+    public let syncPageSize: UInt
+
+    init(errorHandler: @escaping DataStoreErrorHandler,
+         conflictHandler: @escaping DataStoreConflictHandler,
+         syncInterval: UInt,
+         syncMaxRecords: UInt,
+         syncPageSize: UInt) {
+        self.errorHandler = errorHandler
+        self.conflictHandler = conflictHandler
+        self.syncInterval = syncInterval
+        self.syncMaxRecords = syncMaxRecords
+        self.syncPageSize = syncPageSize
+    }
+
+}
+
+extension DataStoreConfiguration {
+
+    public static let defaultSyncInterval: UInt = 1_440
```

Comment:
1440 = 24 * 60 = every 24 minutes?

Reply:
good catch, I think the configuration demands more documentation. That interval is in minutes. In JS that's not very clear either, unless you look at the source code. Good opportunity to improve it - should we adopt a `TimeInterval` approach, with `.minutes(60)`, `.hours(24)`, etc?

Reply:
In Java, we have this awesome [`TimeUnit`](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/TimeUnit.html) conversion utility, backed into the core library since Java 7. Is there not a Swift equivalent?

Reply:
Based on the most recent changes, looks like we've adopted a TimeInterval approach, and looks good to me!. resolving.